app/controllers/distributions_controller.rb

```diff
@@ -57,7 +57,9 @@ def index
 
     respond_to do |format|
       format.html
-      format.csv { send_data Distribution.generate_csv(@distributions, @items.collect(&:name).sort), filename: "Distributions-#{Time.zone.today}.csv" }
+      format.csv do
+        send_data Exports::ExportDistributionsCSVService.new(distribution_ids: @distributions.map(&:id)).generate_csv, filename: "Distributions-#{Time.zone.today}.csv"
```

Comment:
Removed the class method `generate_csv` since I think it complicates and makes it unclear how or what service object it is using.